packages/python-packages/api-stub-generator/apistub/_stub_generator.py

```diff
@@ -0,0 +1,216 @@
+        logging.debug("Modules to generate tokens: {}".format(modules))
+        # load all modules and parse them recursively
+        for m in modules:
+            logging.debug("Importing module {}".format(m))
+            module_obj = importlib.import_module(m)
+            module_dict[m] = ModuleNode(m, module_obj, nodeindex)
+
+        # Create navigation info to navigate within APIreview tool
+        navigation = Navigation(package_name, None)
+        navigation.set_tag(NavigationTag(Kind.type_package))
+        apiview.add_navigation(navigation)
+
+        # Generate tokens
+        modules = module_dict.keys()
+        for m in modules:
+            # Generate and add token to APIView
+            logging.debug("Generating tokens for module {}".format(m))
+            module_dict[m].generate_tokens(apiview)
+            # Add navigation info for this modules. navigation info is used to build tree panel in API tool
+            module_nav = module_dict[m].get_navigation()
+            if module_nav:
+                navigation.add_child(module_nav)
+        return apiview
+
+    def _extract_wheel(self):
+        """Extract the wheel into out dir and return root path to azure root directory in package
+        """
+        file_name, _ = os.path.splitext(os.path.basename(self.pkg_path))
+        temp_pkg_dir = os.path.join(self.temp_path, file_name)
+        if os.path.exists(temp_pkg_dir):
+            logging.debug(
+                "Cleaning up existing temp directory: {}".format(temp_pkg_dir)
+            )
+            shutil.rmtree(temp_pkg_dir)
+        os.mkdir(temp_pkg_dir)
+
+        logging.debug(
+            "Extracting {0} to directory {1}".format(self.pkg_path, temp_pkg_dir)
+        )
+        zip_file = zipfile.ZipFile(self.pkg_path)
+        zip_file.extractall(temp_pkg_dir)
+        logging.debug("Extracted package files into temp path")
+        return temp_pkg_dir
+
+    def _parse_pkg_name(self):
+        file_name = os.path.basename(self.pkg_path)
+        whl_name, extn = os.path.splitext(file_name)
+        if extn[1:] not in ["whl", "zip"]:
+            raise ValueError(
+                "Invalid type of package. API view parser expects wheel or sdist package"
+            )
+
+        filename_parts = whl_name.split("-")
+        pkg_name = filename_parts[0].replace("_", "-")
+        version = filename_parts[1]
+        return pkg_name, version
+
+    def _install_package(self, pkg_name):
+        # Uninstall the package and reinstall it to parse so inspect can get members in package
+        # We don't want to force reinstall to avoid reinstalling other dependent packages
+        commands = [sys.executable, "-m", "pip", "uninstall", pkg_name, "--yes"]
+        check_call(commands)
+        commands = [sys.executable, "-m", "pip", "install", self.pkg_path]
+        check_call(commands)
+
+
+class NodeIndex:
+    """Maintains name to navigation ID"""
+
+    index = {}
```

Comment:
By having this as a class attribute, any value you add will be shared by every instance on `NodeIndex` - is that what you intended?
Example:
```
>>> class Index:
...     index = {}
...     def add(self, name, value):
...         self.index[name] = value
...
>>> a = Index()
>>> a.add("foo1", "bar1")
>>> a.index
{'foo1': 'bar1'}
>>> Index.index
{'foo1': 'bar1'}
>>> b = Index()
>>> b.index
{'foo1': 'bar1'}
```